Parser/Parser.cs

```diff
@@ -61,6 +62,17 @@ NBlock Block ()
       return vars.ToArray ();
    }
 
+    NConstDecl[] ConstDecl () {
+      List<NConstDecl> vars = new ();
+      while (!Peek (VAR)) {
+         var name = Expect (IDENT);
+         Expect (EQ);
+         var val = mToken;
+         if (Match (mToken.Kind) && Match (SEMI)) { vars.Add (new NConstDecl (name, new NLiteral(val))); } 

```

Comment:
You are using Match (mToken.Kind) which will always return true as inside Match method you are check with mToken.Kind. You should expect token whose kind starts with L_.